pkg/core/interop/context.go

```diff
@@ -140,8 +141,14 @@ type Function struct {
 // Method is a signature for a native method.
 type Method = func(ic *Context, args []stackitem.Item) stackitem.Item
 
-// MethodAndPrice is a native-contract method descriptor.
+// MethodAndPrice is a generic hardfork-independent native contract method descriptor.
 type MethodAndPrice struct {
+	HFSpecificMethodAndPrice
+	ActiveFrom *config.Hardfork
```

Comment:
I'm not sure why are you keeping all of them this way. You can have the one and only proper contract version at the current height and then just switch it when HF happens.

Historic calls maybe? These do require some dances.

Reply:
> You can have the one and only proper contract version at the current height

That was an early version of this PR. And yes, due to the historic calls. To manage single contract version at a time, we need to initialize it every time when new historic call is processed (which I suppose is far from the best solution) or we need some kind of cache. So if we're to implement this cache anyway, then it's better to initialize it once during the node start in order not to affect the node performance during blocks processing. And, as discussed, we don't have a lot of natives/hardforks, so the cache size is not a problem for us (and we store pointers in cache anyway, as a result the most part of CS structures are shared between hardforks).

But that's just my thoughts. We can get back to `one and only proper contract version`.